frontend/src/metabase/visualizations/visualizations/PieChart/chart-definition.ts

```diff
@@ -113,11 +113,15 @@ export const PIE_CHART_DEFINITION: VisualizationDefinition = {
     }),
     "pie.rows": {
       hidden: true,
-      getValue: (rawSeries, settings) => {
-        return getPieRows(rawSeries, settings, (value, options) =>
-          String(formatValue(value, options)),
-        );
-      },
+      getValue: _.memoize(
+        (rawSeries, settings) => {
+          return getPieRows(rawSeries, settings, (value, options) =>
+            String(formatValue(value, options)),
+          );
+        },
+        (rawSeries, settings) =>
+          JSON.stringify(rawSeries) + JSON.stringify(settings),
```

Comment:
`rawSeries` can be a huge dataset potentially. I think we can use narrower data about the query run and reduce the number of viz settings keys as well to prevent recomputation on irrelevant settings change. Wdyt?

Reply:
Good call, [`f7c6845`](https://github.com/metabase/metabase/pull/48670/commits/f7c684541cff501738a09ce365bf0814b7a3a169) should address this, please take a look